pkg/apis/hive/v1/hiveconfig_types.go

```diff
@@ -194,6 +214,67 @@ type ManageDNSAzureConfig struct {
 	ResourceGroupName string `json:"resourceGroupName"`
 }
 
+// ControllerGoroutinesConfig contains number of goroutines for each controller
+type ControllerGoroutinesConfig struct {
+	// Default specifies default number of goroutines for each Controller
+	// +optional
+	Default *int `json:"default,omitempty"`
```

Comment:
All `int` fields in the API should use `int32` instead.
```suggestion
	Default *int32 `json:"default,omitempty"`
```